web/src/routes/(user)/people/[personId]/+page.svelte

```diff
@@ -91,8 +90,108 @@
       });
     }
   };
+
+  const handleMergeSameFace = async () => {
+    showMergeModal = false;
+    try {
+      await api.personApi.mergePerson({
+        id: personMerge2.id,
+        mergePersonDto: { ids: [personMerge1.id] },
+      });
+
+      people = people.filter((person: PersonResponseDto) => person.id !== personMerge1.id);
+
+      if (personMerge2.name != personName) {
+        ChangeName();
+        invalidateAll();
+      } else {
+        goto(`${AppRoute.PEOPLE}/${personMerge2.id}`, { replaceState: true });
+        notificationController.show({
+          message: 'Merge faces succesfully',
+          type: NotificationType.Info,
+        });
+      }
+    } catch (error) {
+      handleError(error, 'Unable to save name');
+    }
+  };
+
+  const ChangeName = async () => {
+    try {
+      isEditingName = false;
+
+      const { data: updatedPerson } = await api.personApi.updatePerson({
+        id: data.person.id,
+        personUpdateDto: { name: personName },
+      });
+
+      people = people.map((person: PersonResponseDto) => {
+        if (person.id === updatedPerson.id) {
+          return updatedPerson;
+        }
+        return person;
+      });
+
+      notificationController.show({
+        message: 'Change name succesfully',
+        type: NotificationType.Info,
+      });
+    } catch (error) {
+      handleError(error, 'Unable to save name');
+    }
+  };
+
+  const handleNameChange = async (name: string) => {
+    personName = name;
+    let detectnew = false;
+    if (data.person.name != personName) {
+      for (const person of people) {
+        if (person.name === personName && person.id !== data.person.id) {
+          personMerge2 = person;
+          detectnew = true;
+          break;
+        }
+      }
+      if (!detectnew) {
+        await ChangeName();
+        data.person.name = personName;
+        people.map((person: PersonResponseDto) => {
+          if (person.id === data.person.id) {
+            person.name = personName;
+          }
+          return person;
+        });
+      } else {
+        personMerge1 = data.person;
+        potentialMergePeople = people
+          .filter(
+            (person: PersonResponseDto) =>
+              personName === person.name && person.id !== personMerge2.id && person.id !== personMerge1.id,
+          )
+          .slice(0, 3);
+        showMergeModal = true;
+      }
+    }
+  };
 </script>
 
+{#if showMergeModal}
+  <FullScreenModal on:clickOutside={() => (showMergeModal = false)}>
+    <SuggestMerge
+      bind:personMerge1
+      bind:personMerge2
+      bind:potentialMergePeople
+      on:close={() => (showMergeModal = false)}
+      on:differentFaces={async () => {
+        showMergeModal = false;
+        isEditingName = false;
```

Comment:
Seems like these should always be set to false once you get to `changeName()`, so maybe they can be moved in there instead.